# Global Food Prices: 2019–2026

## From Pre-COVID to 2026: How Have Global Food Prices Changed?

Source: World Food Programme (WFP)

Project Goal: Analyse WFP food-price data from 2019 to 2026 to understand how food prices changed before, during and after the COVID-19 period, and identify meaningful differences across countries, markets and food commodities.

1. Import Libraries


In [ ]:
import pandas as pd


2. Initial Dataset Inspection — 2019

In [ ]:
df2019 = pd.read_csv("wfp_food_prices_global_2019.csv")

In [ ]:
df2019.head()

,countryiso3,date,admin1,admin2,market,market_id,latitude,longitude,category,commodity,commodity_id,unit,priceflag,pricetype,currency,price,usdprice
0,AFG,2019-01-15,Badakhshan,Faiz Abad,Fayzabad,266,37.12,70.58,cereals and tubers,Bread,55,KG,actual,Retail,AFN,50.00,0.67
1,AFG,2019-01-15,Badakhshan,Faiz Abad,Fayzabad,266,37.12,70.58,cereals and tubers,Rice (low quality),145,KG,actual,Retail,AFN,60.25,0.80
2,AFG,2019-01-15,Badakhshan,Faiz Abad,Fayzabad,266,37.12,70.58,cereals and tubers,Wheat,84,KG,actual,Retail,AFN,22.75,0.30
3,AFG,2019-01-15,Badakhshan,Faiz Abad,Fayzabad,266,37.12,70.58,non-food,Exchange rate,305,USD/LCU,actual,Retail,AFN,75.22,1.00
4,AFG,2019-01-15,Badakhshan,Faiz Abad,Fayzabad,266,37.12,70.58,non-food,Fuel (diesel),284,L,actual,Retail,AFN,56.00,0.75


In [ ]:
df2019.shape

(220843, 17)

In [ ]:
df2019.columns

Index(['countryiso3', 'date', 'admin1', 'admin2', 'market', 'market_id',
       'latitude', 'longitude', 'category', 'commodity', 'commodity_id',
       'unit', 'priceflag', 'pricetype', 'currency', 'price', 'usdprice'],
      dtype='object')

In [ ]:
df2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 220843 entries, 0 to 220842
Data columns (total 17 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   countryiso3   220843 non-null  object 
 1   date          220843 non-null  object 
 2   admin1        218689 non-null  object 
 3   admin2        218689 non-null  object 
 4   market        220843 non-null  object 
 5   market_id     220843 non-null  int64  
 6   latitude      218689 non-null  float64
 7   longitude     218689 non-null  float64
 8   category      220843 non-null  object 
 9   commodity     220843 non-null  object 
 10  commodity_id  220843 non-null  int64  
 11  unit          220843 non-null  object 
 12  priceflag     220843 non-null  object 
 13  pricetype     220843 non-null  object 
 14  currency      220843 non-null  object 
 15  price         220843 non-null  float64
 16  usdprice      220843 non-null  float64
dtypes: float64(4), int64(2), object(11)
memory usage

In [ ]:
df2019.isnull().sum()

,0
countryiso3,0
date,0
admin1,2154
admin2,2154
market,0
market_id,0
latitude,2154
longitude,2154
category,0
commodity,0


In [ ]:
df2019.duplicated().sum()

np.int64(0)

In [ ]:
# Check the earliest and latest date in the dataset
print("DATE RANGE:")
print(df2019["date"].min(), "to", df2019["date"].max())

DATE RANGE:
2019-01-15 to 2019-12-15


In [ ]:
# Count the number of unique countries
print("\nNUMBER OF COUNTRIES:")
print(df2019["countryiso3"].nunique())


NUMBER OF COUNTRIES:
89


In [ ]:
# Count and display the food categories
print("\nFOOD CATEGORIES:")
print(df2019["category"].nunique())
print(df2019["category"].value_counts())


FOOD CATEGORIES:
8
category
cereals and tubers       79386
vegetables and fruits    30804
pulses and nuts          28513
meat, fish and eggs      25479
non-food                 19435
miscellaneous food       14638
oil and fats             14568
milk and dairy            8020
Name: count, dtype: int64


In [ ]:
# Count the number of unique commodities
print("\nNUMBER OF COMMODITIES:")
print(df2019["commodity"].nunique())


NUMBER OF COMMODITIES:
459


In [ ]:
# Display the 20 most frequently recorded commodities
print("\nTOP 20 COMMODITIES:")
print(df2019["commodity"].value_counts().head(20))


TOP 20 COMMODITIES:
commodity
Millet                    5845
Rice (imported)           5576
Sugar                     5527
Maize (white)             5036
Rice (local)              4800
Wheat flour               4384
Maize                     4289
Tomatoes                  4246
Onions                    3791
Salt                      3741
Rice                      3690
Oil (vegetable)           3526
Fuel (diesel)             3503
Eggs                      3441
Potatoes                  3311
Oil (palm)                3215
Sorghum                   3167
Groundnuts (shelled)      2708
Meat (beef)               2675
Fuel (petrol-gasoline)    2670
Name: count, dtype: int64


In [ ]:
#Shows how many observations are retail, wholesale, or other price types. This will help us decide which type is appropriate for our analysis.
# Count observations by price type
print("\nPRICE TYPES:")
print(df2019["pricetype"].value_counts())


PRICE TYPES:
pricetype
Retail       205679
Wholesale     15164
Name: count, dtype: int64


In [ ]:
#Shows how the WFP observations are classified, such as actual or aggregate, so we can understand the nature of the recorded prices.
# Count observations by price flag
print("\nPRICE FLAGS:")
print(df2019["priceflag"].value_counts())


PRICE FLAGS:
priceflag
actual              213300
aggregate             6085
actual,aggregate      1458
Name: count, dtype: int64


In [ ]:
# Count the different measurement units
print("\nNUMBER OF UNITS:")
print(df2019["unit"].nunique())


NUMBER OF UNITS:
122


3. Compare the 2019–2026 Datasets
Purpose: Compare the yearly datasets before merging them to ensure that their structure, coverage and variables are compatible.

In [ ]:
# Storing the file name for each seperate years
file_names = {
    2019: "wfp_food_prices_global_2019.csv",
    2020: "wfp_food_prices_global_2020.csv",
    2021: "wfp_food_prices_global_2021.csv",
    2022: "wfp_food_prices_global_2022.csv",
    2023: "wfp_food_prices_global_2023.csv",
    2024: "wfp_food_prices_global_2024.csv",
    2025: "wfp_food_prices_global_2025.csv",
    2026: "wfp_food_prices_global_2026.csv"
}

In [ ]:
# Load all eight CSV files into separate DataFrames
dfs = {}

for year, file in file_names.items():
    dfs[year] = pd.read_csv(file)

print("All files loaded successfully!")

All files loaded successfully!


In [ ]:
# Display the years that were successfully loaded
print(dfs.keys())

dict_keys([2019, 2020, 2021, 2022, 2023, 2024, 2025, 2026])


In [ ]:
# Display the number of rows and columns for every year
for year, df in dfs.items():
    print(year, ":", df.shape)

2019 : (220843, 17)
2020 : (435241, 17)
2021 : (488775, 17)
2022 : (497042, 17)
2023 : (484984, 17)
2024 : (448410, 17)
2025 : (441377, 17)
2026 : (212905, 17)


In [ ]:
# Compare the column names across all eight datasets
first_columns = dfs[2019].columns

for year, df in dfs.items():
    print(year, list(df.columns) == list(first_columns))

2019 True
2020 True
2021 True
2022 True
2023 True
2024 True
2025 True
2026 True


In [ ]:
# Check the earliest and latest date available in each year
for year, df in dfs.items():
    print(year, ":", df["date"].min(), "to", df["date"].max())

2019 : 2019-01-15 to 2019-12-15
2020 : 2020-01-15 to 2020-12-15
2021 : 2021-01-15 to 2021-12-15
2022 : 2022-01-15 to 2022-12-15
2023 : 2023-01-15 to 2023-12-15
2024 : 2024-01-15 to 2024-12-15
2025 : 2025-01-15 to 2025-12-15
2026 : 2026-01-15 to 2026-08-15


In [ ]:
# Count the number of unique countries in each year
for year, df in dfs.items():
    print(year, ":", df["countryiso3"].nunique(), "countries")

2019 : 89 countries
2020 : 89 countries
2021 : 85 countries
2022 : 87 countries
2023 : 73 countries
2024 : 72 countries
2025 : 73 countries
2026 : 70 countries


In [ ]:
# Count the number of unique commodities in each year
for year, df in dfs.items():
    print(year, ":", df["commodity"].nunique(), "commodities")

2019 : 459 commodities
2020 : 660 commodities
2021 : 679 commodities
2022 : 652 commodities
2023 : 643 commodities
2024 : 661 commodities
2025 : 665 commodities
2026 : 632 commodities


In [ ]:
# Count the number of unique markets in each year
for year, df in dfs.items():
    print(year, ":", df["market_id"].nunique(), "markets")

2019 : 1799 markets
2020 : 2748 markets
2021 : 2717 markets
2022 : 2795 markets
2023 : 3054 markets
2024 : 2961 markets
2025 : 3079 markets
2026 : 2392 markets


In [ ]:
# Count the different price types in each year
#Shows how many observations are Retail, Wholesale, or other price types in each year
for year, df in dfs.items():
    print("\n", year)
    print(df["pricetype"].value_counts())


 2019
pricetype
Retail       205679
Wholesale     15164
Name: count, dtype: int64

 2020
pricetype
Retail       406700
Wholesale     28541
Name: count, dtype: int64

 2021
pricetype
Retail       456164
Wholesale     32611
Name: count, dtype: int64

 2022
pricetype
Retail       468075
Wholesale     28967
Name: count, dtype: int64

 2023
pricetype
Retail       458217
Wholesale     26767
Name: count, dtype: int64

 2024
pricetype
Retail       423580
Wholesale     24830
Name: count, dtype: int64

 2025
pricetype
Retail       422296
Wholesale     19081
Name: count, dtype: int64

 2026
pricetype
Retail       204772
Wholesale      8133
Name: count, dtype: int64


In [ ]:
# Check how price observations are classified in each year
for year, df in dfs.items():
    print("\n", year)
    print(df["priceflag"].value_counts())


 2019
priceflag
actual              213300
aggregate             6085
actual,aggregate      1458
Name: count, dtype: int64

 2020
priceflag
actual              290098
aggregate           141756
actual,aggregate      3387
Name: count, dtype: int64

 2021
priceflag
actual              323893
aggregate           163346
actual,aggregate      1536
Name: count, dtype: int64

 2022
priceflag
actual              339865
aggregate           155869
actual,aggregate      1308
Name: count, dtype: int64

 2023
priceflag
actual              312295
aggregate           171243
actual,aggregate      1446
Name: count, dtype: int64

 2024
priceflag
actual              331031
aggregate           115618
actual,aggregate      1761
Name: count, dtype: int64

 2025
priceflag
actual              314807
aggregate           121728
actual,aggregate      4842
Name: count, dtype: int64

 2026
priceflag
actual              144216
aggregate            64339
actual,aggregate      4350
Name: count, dtype: int64


In [ ]:
# Count missing values in each column for every year
for year, df in dfs.items():
    print("\nMISSING VALUES -", year)
    print(df.isnull().sum())


MISSING VALUES - 2019
countryiso3        0
date               0
admin1          2154
admin2          2154
market             0
market_id          0
latitude        2154
longitude       2154
category           0
commodity          0
commodity_id       0
unit               0
priceflag          0
pricetype          0
currency           0
price              0
usdprice           0
dtype: int64

MISSING VALUES - 2020
countryiso3        0
date               0
admin1          1464
admin2          1464
market             0
market_id          0
latitude        1464
longitude       1464
category           0
commodity          0
commodity_id       0
unit               0
priceflag          0
pricetype          0
currency           0
price              0
usdprice         287
dtype: int64

MISSING VALUES - 2021
countryiso3        0
date               0
admin1          1340
admin2          1340
market             0
market_id          0
latitude        1340
longitude       1340
category           0
co

In [ ]:
# Count the number of different measurement units in each year
for year, df in dfs.items():
    print(year, ":", df["unit"].nunique(), "different units")

2019 : 122 different units
2020 : 141 different units
2021 : 137 different units
2022 : 139 different units
2023 : 126 different units
2024 : 118 different units
2025 : 117 different units
2026 : 109 different units


In [ ]:
# Display the 20 most common measurement units in each year
for year, df in dfs.items():
    print("\n", year)
    print(df["unit"].value_counts().head(20))


 2019
unit
KG            154824
L              19922
100 KG          3821
Head            2436
Unit            2169
Day             2005
30 pcs          1870
50 KG           1622
400 G           1609
10 tablets      1454
MT              1444
500 G           1424
250 G           1030
90 KG           1029
10 pcs          1022
2.5 KG          1010
160 G            965
200 G            955
USD/LCU          942
Pound            797
Name: count, dtype: int64

 2020
unit
KG            327738
L              28021
100 KG          8932
Unit            7435
Head            5017
Day             3887
10 pcs          2666
50 KG           2520
500 G           2451
30 pcs          1863
Packet          1740
400 G           1714
250 G           1707
100 ML          1570
10 tablets      1449
USD/LCU         1380
25 KG           1357
1.1 KG          1348
200 G           1344
Heap            1175
Name: count, dtype: int64

 2021
unit
KG            367967
L              31612
Unit           12408
100 KG   

In [ ]:
# Check for completely duplicated rows in each year
for year, df in dfs.items():
    print(year, ":", df.duplicated().sum(), "duplicate rows")

2019 : 0 duplicate rows
2020 : 0 duplicate rows
2021 : 0 duplicate rows
2022 : 0 duplicate rows
2023 : 0 duplicate rows
2024 : 0 duplicate rows
2025 : 0 duplicate rows
2026 : 0 duplicate rows


In [ ]:
# Check whether all years have exactly the same column names
columns_2019 = list(dfs[2019].columns)

for year, df in dfs.items():
    print(year, list(df.columns) == columns_2019)

2019 True
2020 True
2021 True
2022 True
2023 True
2024 True
2025 True
2026 True


4. Investigate Missing USD Prices
**Understand why some observations have missing USD prices before deciding how they should be handled during preprocessing.**

In [ ]:
# Check which currencies have missing USD prices in each year
for year, df in dfs.items():
    missing_usd = df[df["usdprice"].isna()]

    print("\nYEAR:", year)
    print("Missing USD prices:", len(missing_usd))
    print(missing_usd["currency"].value_counts())


YEAR: 2019
Missing USD prices: 0
Series([], Name: count, dtype: int64)

YEAR: 2020
Missing USD prices: 287
currency
SLS    287
Name: count, dtype: int64

YEAR: 2021
Missing USD prices: 426
currency
SLS    426
Name: count, dtype: int64

YEAR: 2022
Missing USD prices: 487
currency
SLS    487
Name: count, dtype: int64

YEAR: 2023
Missing USD prices: 725
currency
SLS    725
Name: count, dtype: int64

YEAR: 2024
Missing USD prices: 986
currency
SLS    986
Name: count, dtype: int64

YEAR: 2025
Missing USD prices: 1400
currency
SLS    1010
PRB     390
Name: count, dtype: int64

YEAR: 2026
Missing USD prices: 579
currency
SLS    505
PRB     74
Name: count, dtype: int64


In [ ]:
# Check which countries have missing USD prices
for year, df in dfs.items():
    missing_usd = df[df["usdprice"].isna()]

    print("\nYEAR:", year)
    print(missing_usd["countryiso3"].value_counts())


YEAR: 2019
Series([], Name: count, dtype: int64)

YEAR: 2020
countryiso3
SOM    287
Name: count, dtype: int64

YEAR: 2021
countryiso3
SOM    426
Name: count, dtype: int64

YEAR: 2022
countryiso3
SOM    487
Name: count, dtype: int64

YEAR: 2023
countryiso3
SOM    725
Name: count, dtype: int64

YEAR: 2024
countryiso3
SOM    986
Name: count, dtype: int64

YEAR: 2025
countryiso3
SOM    1010
MDA     390
Name: count, dtype: int64

YEAR: 2026
countryiso3
SOM    505
MDA     74
Name: count, dtype: int64


4. Data Preprocessing and Cleaning

Purpose:
Prepare the 2019–2026 WFP food-price datasets for reliable analysis by standardising variables, handling data-quality issues, applying appropriate filters, and creating an analysis-ready master dataset.

4.1 Date Standardisation

In [ ]:
# Convert the date column into proper datetime format
for year, df in dfs.items():
    df["date"] = pd.to_datetime(df["date"])

In [ ]:
# Check that the date column is now in datetime format
for year, df in dfs.items():
    print(year, ":", df["date"].dtype)

2019 : datetime64[ns]
2020 : datetime64[ns]
2021 : datetime64[ns]
2022 : datetime64[ns]
2023 : datetime64[ns]
2024 : datetime64[ns]
2025 : datetime64[ns]
2026 : datetime64[ns]


In [ ]:
# Check the earliest and latest date for each year
for year, df in dfs.items():
    print(year, ":", df["date"].min(), "to", df["date"].max())

2019 : 2019-01-15 00:00:00 to 2019-12-15 00:00:00
2020 : 2020-01-15 00:00:00 to 2020-12-15 00:00:00
2021 : 2021-01-15 00:00:00 to 2021-12-15 00:00:00
2022 : 2022-01-15 00:00:00 to 2022-12-15 00:00:00
2023 : 2023-01-15 00:00:00 to 2023-12-15 00:00:00
2024 : 2024-01-15 00:00:00 to 2024-12-15 00:00:00
2025 : 2025-01-15 00:00:00 to 2025-12-15 00:00:00
2026 : 2026-01-15 00:00:00 to 2026-08-15 00:00:00


4.2 Creating a year Variable

In [ ]:
# Extract the year from the date column
for year, df in dfs.items():
    df["year"] = df["date"].dt.year

In [ ]:
# Check the years stored in each dataset
for year, df in dfs.items():
    print(year, ":", df["year"].unique())

2019 : [2019]
2020 : [2020]
2021 : [2021]
2022 : [2022]
2023 : [2023]
2024 : [2024]
2025 : [2025]
2026 : [2026]


In [ ]:
# Display the first five dates and their corresponding year
dfs[2019][["date", "year"]].head()

,date,year
0,2019-01-15,2019
1,2019-01-15,2019
2,2019-01-15,2019
3,2019-01-15,2019
4,2019-01-15,2019


4.3  Examining Price Types

In [ ]:
# Check price types across all years
for year, df in dfs.items():
    print("\n", year)
    print(df["pricetype"].value_counts())


 2019
pricetype
Retail       205679
Wholesale     15164
Name: count, dtype: int64

 2020
pricetype
Retail       406700
Wholesale     28541
Name: count, dtype: int64

 2021
pricetype
Retail       456164
Wholesale     32611
Name: count, dtype: int64

 2022
pricetype
Retail       468075
Wholesale     28967
Name: count, dtype: int64

 2023
pricetype
Retail       458217
Wholesale     26767
Name: count, dtype: int64

 2024
pricetype
Retail       423580
Wholesale     24830
Name: count, dtype: int64

 2025
pricetype
Retail       422296
Wholesale     19081
Name: count, dtype: int64

 2026
pricetype
Retail       204772
Wholesale      8133
Name: count, dtype: int64


4.4 Examining Price Flags

In [ ]:
# Check price flags across all years
for year, df in dfs.items():
    print("\n", year)
    print(df["priceflag"].value_counts())


 2019
priceflag
actual              213300
aggregate             6085
actual,aggregate      1458
Name: count, dtype: int64

 2020
priceflag
actual              290098
aggregate           141756
actual,aggregate      3387
Name: count, dtype: int64

 2021
priceflag
actual              323893
aggregate           163346
actual,aggregate      1536
Name: count, dtype: int64

 2022
priceflag
actual              339865
aggregate           155869
actual,aggregate      1308
Name: count, dtype: int64

 2023
priceflag
actual              312295
aggregate           171243
actual,aggregate      1446
Name: count, dtype: int64

 2024
priceflag
actual              331031
aggregate           115618
actual,aggregate      1761
Name: count, dtype: int64

 2025
priceflag
actual              314807
aggregate           121728
actual,aggregate      4842
Name: count, dtype: int64

 2026
priceflag
actual              144216
aggregate            64339
actual,aggregate      4350
Name: count, dtype: int64


4.5 Examining Food Categories

In [ ]:
# Check food categories across all years
for year, df in dfs.items():
    print("\n", year)
    print(df["category"].value_counts())


 2019
category
cereals and tubers       79386
vegetables and fruits    30804
pulses and nuts          28513
meat, fish and eggs      25479
non-food                 19435
miscellaneous food       14638
oil and fats             14568
milk and dairy            8020
Name: count, dtype: int64

 2020
category
cereals and tubers       123705
vegetables and fruits     91537
meat, fish and eggs       70676
pulses and nuts           40292
non-food                  36515
miscellaneous food        32634
oil and fats              29703
milk and dairy            10179
Name: count, dtype: int64

 2021
category
cereals and tubers       134006
vegetables and fruits    102616
meat, fish and eggs       76982
pulses and nuts           46942
non-food                  39510
miscellaneous food        39323
oil and fats              37306
milk and dairy            12090
Name: count, dtype: int64

 2022
category
cereals and tubers       133694
vegetables and fruits    104230
meat, fish and eggs       78407
pu

In [ ]:
# Remove non-food observations from all yearly datasets
for year, df in dfs.items():
    dfs[year] = df[df["category"] != "non-food"].copy()

In [ ]:
# Check that non-food observations have been removed
for year, df in dfs.items():
    print(year, ":", df["category"].unique())

2019 : ['cereals and tubers' 'milk and dairy' 'miscellaneous food' 'oil and fats'
 'meat, fish and eggs' 'pulses and nuts' 'vegetables and fruits']
2020 : ['cereals and tubers' 'miscellaneous food' 'oil and fats'
 'pulses and nuts' 'milk and dairy' 'meat, fish and eggs'
 'vegetables and fruits']
2021 : ['cereals and tubers' 'miscellaneous food' 'oil and fats'
 'pulses and nuts' 'meat, fish and eggs' 'milk and dairy'
 'vegetables and fruits']
2022 : ['cereals and tubers' 'miscellaneous food' 'oil and fats'
 'pulses and nuts' 'meat, fish and eggs' 'milk and dairy'
 'vegetables and fruits']
2023 : ['cereals and tubers' 'miscellaneous food' 'oil and fats'
 'pulses and nuts' 'meat, fish and eggs' 'milk and dairy'
 'vegetables and fruits']
2024 : ['cereals and tubers' 'miscellaneous food' 'oil and fats'
 'pulses and nuts' 'meat, fish and eggs' 'milk and dairy'
 'vegetables and fruits']
2025 : ['cereals and tubers' 'miscellaneous food' 'oil and fats'
 'pulses and nuts' 'meat, fish and eggs' '

4.6 Measurement Units

In [ ]:
# Check the most common measurement units across all years
for year, df in dfs.items():
    print("\n", year)
    print(df["unit"].value_counts().head(15))


 2019
unit
KG        154401
L          11858
100 KG      3821
Head        2436
Unit        2169
400 G       1609
30 pcs      1486
MT          1444
500 G       1424
50 KG       1204
90 KG       1029
2.5 KG      1010
250 G       1009
160 G        965
200 G        955
Name: count, dtype: int64

 2020
unit
KG        325878
L          17068
100 KG      8932
Unit        5759
Head        5017
500 G       2322
10 pcs      2058
400 G       1714
30 pcs      1693
50 KG       1480
1.1 KG      1348
Packet      1307
200 G       1228
Heap        1157
3.5 KG      1134
Name: count, dtype: int64

 2021
unit
KG        365568
L          19906
100 KG      8577
Unit        8052
Head        4977
500 G       3112
10 pcs      2538
Heap        2286
30 pcs      1900
400 G       1870
50 KG       1628
200 G       1592
Bunch       1378
3.5 KG      1361
5 L         1193
Name: count, dtype: int64

 2022
unit
KG        375619
L          20473
100 KG      7949
Unit        6420
Head        6396
10 pcs      2947
500 G  

In [ ]:
# Count the number of different measurement units in each year
for year, df in dfs.items():
    print(year, ":", df["unit"].nunique(), "different units")

2019 : 106 different units
2020 : 114 different units
2021 : 112 different units
2022 : 116 different units
2023 : 104 different units
2024 : 94 different units
2025 : 92 different units
2026 : 80 different units


4.7 Handling Missing USD Prices

Missing USD prices were retained in the master dataset because the original local-currency price values are available. No estimated exchange rates were introduced. For analyses requiring USD-based comparisons, observations with missing usdprice will be excluded.

4.8  Basic Price Validation

In [ ]:
# Check for zero or negative prices
for year, df in dfs.items():
    print(year, ":", (df["price"] <= 0).sum())

2019 : 0
2020 : 0
2021 : 0
2022 : 0
2023 : 0
2024 : 0
2025 : 0
2026 : 0


In [ ]:
# Check for zero or negative USD prices
for year, df in dfs.items():
    print(year, ":", (df["usdprice"].fillna(0) <= 0).sum())

2019 : 0
2020 : 210
2021 : 318
2022 : 364
2023 : 622
2024 : 866
2025 : 1280
2026 : 519


In [ ]:
# Check actual non-missing USD prices that are zero or negative
for year, df in dfs.items():
    invalid_usd = df["usdprice"].notna() & (df["usdprice"] <= 0)
    print(year, ":", invalid_usd.sum())

2019 : 0
2020 : 0
2021 : 0
2022 : 0
2023 : 0
2024 : 0
2025 : 0
2026 : 0


In [ ]:
# Check actual zero and negative local prices
for year, df in dfs.items():
    invalid_price = df["price"] <= 0
    print("\n", year)
    print(df.loc[invalid_price, ["commodity", "unit", "price", "currency"]].head())


 2019
Empty DataFrame
Columns: [commodity, unit, price, currency]
Index: []

 2020
Empty DataFrame
Columns: [commodity, unit, price, currency]
Index: []

 2021
Empty DataFrame
Columns: [commodity, unit, price, currency]
Index: []

 2022
Empty DataFrame
Columns: [commodity, unit, price, currency]
Index: []

 2023
Empty DataFrame
Columns: [commodity, unit, price, currency]
Index: []

 2024
Empty DataFrame
Columns: [commodity, unit, price, currency]
Index: []

 2025
Empty DataFrame
Columns: [commodity, unit, price, currency]
Index: []

 2026
Empty DataFrame
Columns: [commodity, unit, price, currency]
Index: []


RESULT: All observations across the 2019–2026 datasets have positive local-currency prices. No zero or negative price values were found, so no records were removed for invalid local prices.

4.9 Combining the 8 datasets

In [ ]:
# Combine all cleaned yearly datasets
master_df = pd.concat(dfs.values(), ignore_index=True)

print("Master dataset shape:", master_df.shape)

Master dataset shape: (2945227, 18)


In [ ]:
# Check that all eight years are present
print(master_df["year"].value_counts().sort_index())

year
2019    201408
2020    398726
2021    449265
2022    454668
2023    443467
2024    407882
2025    398183
2026    191628
Name: count, dtype: int64


4.10  Final Validation

In [ ]:
print("Shape:", master_df.shape)
print("Duplicate rows:", master_df.duplicated().sum())
print("Missing values:")
print(master_df.isnull().sum())
print("Years:", sorted(master_df["year"].unique()))

Shape: (2945227, 18)
Duplicate rows: 0
Missing values:
countryiso3        0
date               0
admin1          6640
admin2          6640
market             0
market_id          0
latitude        6640
longitude       6640
category           0
commodity          0
commodity_id       0
unit               0
priceflag          0
pricetype          0
currency           0
price              0
usdprice        4179
year               0
dtype: int64
Years: [np.int32(2019), np.int32(2020), np.int32(2021), np.int32(2022), np.int32(2023), np.int32(2024), np.int32(2025), np.int32(2026)]


Result: The final master dataset contains 2,945,227 observations covering 2019–2026. No duplicate rows were found. The main price, date, country, commodity and category fields contain no missing values. Missing USD prices and geographic fields were retained because they do not prevent the core food-price analysis. All eight years are present in the final dataset.

In [ ]:
# Define the Google Drive folder where the project files are stored
folder = "/content/drive/MyDrive/Global Food Prices 2019-2026"

In [ ]:
import os

folder = "/content/drive/MyDrive/Global Food Prices 2019-2026"

os.makedirs(folder, exist_ok=True)

print("Folder created:", folder)

Folder created: /content/drive/MyDrive/Global Food Prices 2019-2026


In [ ]:
#4.11 Export Clean Dataset
# Save the cleaned master dataset
output_file = f"{folder}/WFP_Food_Prices_2019_2026_Cleaned.csv"

master_df.to_csv(output_file, index=False)

print("Clean dataset saved successfully!")

Clean dataset saved successfully!
